### PR TITLE
add duktape

### PR DESCRIPTION
**Pull request by Gymxo**

### CODE REVIEW

File: duktape.mk [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	$(SED) -i 's/Version:/Version: $(DUKTAPE_VERSION)/g' \
+	$(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/pkgconfig/duktape.pc
+
+
+ifneq ($(wildcard $(BUILD_WORK)/duktape/.build_complete),)
+duktape:
+	@echo "Using previously built duktape."
+else
+duktape: duktape-setup
+	cd $(BUILD_WORK)/duktape && python2 tools/configure.py --output-directory \
```
Comment (TheRealKeto) [inline duktape.mk:27]: You should change this to use python3 aka ``python$(PYTHON_MAJOR_V)``

File: duktape.mk [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	cp -a $(BUILD_STAGE)/duktape/* $(BUILD_BASE)
+	touch $(BUILD_WORK)/duktape/.build_complete
+endif
+
+duktape-package: duktape-stage
+	# duktape.mk Package Structure
+	rm -rf $(BUILD_DIST)/{libduktape206,duktape,duktape-dev}
+	mkdir -p $(BUILD_DIST)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
+	mkdir -p $(BUILD_DIST)/duktape-dev/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/{lib/pkgconfig,include,share/duktape}
+	mkdir -p $(BUILD_DIST)/libduktape206/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib
```
Comment (TheRealKeto) [inline duktape.mk:44]: These can be done in one line.
```makefile
mkdir -p $(1) \
      $(2)
      $(3)
````

File: build_info/duktape.control [hunk 3]
```diff
@@ -0,0 +1,34 @@
+Package: duktape
+Version: @DEB_DUKTAPE_V@
+Architecture: @DEB_ARCH@
+Maintainer: @DEB_MAINTAINER@
```
Comment (TheRealKeto) [inline build_info/duktape.control:4]: Add the author of the package on all your control files below the Maintainer field.

**Review by TheRealKeto — CHANGES_REQUESTED** [review]

File: duktape.mk (hunk 1 above)
Comment (Diatrus) [reply duktape.mk:27]: Just python3 would do fine

File: duktape.mk (hunk 2 above)
Comment (Gymxo) [reply duktape.mk:44]: I usually never do them all in one line

Comment (Gymxo) [reply duktape.mk:44]: Unless Hayden tells me to then I will

File: build_info/duktape.control (hunk 3 above)
Comment (Gymxo) [reply build_info/duktape.control:4]: Also, there is no author for this package
Maybe if you check Debian first before posting

File: duktape.mk (hunk 2 above)
Comment (TheRealKeto) [reply duktape.mk:44]: Doing so reduces repetition; it's easier to read and looks less dumb.

File: duktape.mk (hunk 1 above)
Comment (Gymxo) [reply duktape.mk:27]: I'll test python3 rn

Comment (Gymxo) [reply duktape.mk:27]: the configure file would need to be modified to work with python3

Comment (Gymxo) [reply duktape.mk:27]: Just tested python3 with this and errored

File: duktape.mk (hunk 2 above)
Comment (Diatrus) [reply duktape.mk:44]: Do it in one line. Why run a command more than once

Comment (Gymxo) [reply duktape.mk:44]: Didn't realize you meant the mkdir lol

Comment (Gymxo) [reply duktape.mk]: fixed

**Pull request merged**